.github/scripts/format_dbt_test_failures.py

```diff
@@ -189,6 +295,93 @@ def main() -> None:
         date_today = datetime.datetime.today().strftime("%Y-%m-%d")
         output_filepath = f"qc_test_failures_{date_today}.xlsx"
 
+    failed_test_cache_path = get_failed_test_cache_path(
+        run_results_filepath,
+        manifest_filepath,
+    )
+
+    if os.path.isfile(failed_test_cache_path):
+        print(f"Loading failed tests from cache at {failed_test_cache_path}")
+        failed_tests_by_category = get_failed_tests_by_category_from_file(
+            failed_test_cache_path
+        )
+    else:
+        print(
+            f"Failed test cache not found at {failed_test_cache_path}, "
+            "loading failed tests from Athena"
+        )
+        failed_tests_by_category = get_failed_tests_by_category_from_athena(
+            run_results_filepath,
+            manifest_filepath,
+        )
+        print(f"Saving failed tests to the cache at {failed_test_cache_path}")
+        save_failed_tests_by_category_to_file(
+            failed_tests_by_category,
+            failed_test_cache_path,
+        )
+
+    print("Generating the output workbook")
+    # It's important to use a write-only workbook here because otherwise
+    # the metadata required to store cell info about a large number of failing
+    # tests can cause the process to run out of memory
+    workbook = openpyxl.Workbook(write_only=True)
+    for sheet_name, failed_test_group in failed_tests_by_category.items():
+        print(f"Adding sheet for {sheet_name}")
+        add_sheet_to_workbook(workbook, sheet_name, failed_test_group)
+    workbook.save(output_filepath)
+    print(f"Output workbook saved to {output_filepath}")
+
+
+def get_failed_test_cache_path(
```

Comment:
Just to check my understanding, you're basically just caching the Athena results and using them as long as they exist to speed up testing? Then if they don't exist, query Athena? Goal being to not query Athena every time we build the workbook?

Reply:
Yup, exactly! Otherwise, the queries take up a bunch of time unnecessarily.